src/main/java/Bob.java

```diff
@@ -0,0 +1,338 @@
+
+        // Check if a description exists
+        String[] command_desc = splitCommand[0].split(" ");
+
+
+        return validMatches
+                && splitCommand.length == 3
+                && command_desc.length > 1
+                && command_desc[0].equals("event")
+                && s.indexOf("/from") < s.indexOf("/to") // A valid command has /from before /to
+                && isDate(splitCommand[1])
+                && isDate(splitCommand[2]);
+    }
+
+    private void addEvent(String s) {
+        String[] command = s.split( " /from | /to ");
+        String[] command_desc = command[0].split(" ", 2);
+        String description = command_desc[1];
+        LocalDate start = LocalDate.parse(command[1], format);
+        LocalDate end = LocalDate.parse(command[2], format);
+
+        Event e = new Event(description, start, end);
+        taskList.add(e);
+    }
+
+    // Command: deadline <description> /by <deadline>
+    private boolean isDeadline(String s) {
+        String[] splitCommand = s.split(" /by ");
+        String[] command_desc = splitCommand[0].split(" ", 2);
+
+        return splitCommand.length == 2
+                && command_desc.length == 2
+                && command_desc[0].equals("deadline")
+                && isDate(splitCommand[1]);
+    }
+
+    private void addDeadline(String s) {
+        String[] splitCommand = s.split(" /by ");
+        String[] command_desc = splitCommand[0].split(" ", 2);
+        String description = command_desc[1];
+        LocalDate deadline = LocalDate.parse(splitCommand[1], format);
+        Deadline d = new Deadline(description, deadline);
+        taskList.add(d);
+    }
+    private void addTask(String input) throws BobException {
+        if (isTodo(input)) {
+            addTodo(input);
+        } else if (isEvent(input)) {
+            addEvent(input);
+        } else if (isDeadline(input)) {
+            addDeadline(input);
+        } else { // Invalid command
+            throw new BobException("Sorry :( no valid command was entered");
+        }
+
+        // Get added task
+        Task t = taskList.get(taskList.size() - 1);
+        formattedPrint("I've added a new task!\n" + getTaskDescription(t));
+    }
+
+    private boolean isDelete(String input) {
+        String[] command = input.split(" ");
+        return command.length == 2 && command[0].equals("delete") && isInt(command[1]);
+    }
+
+    private void deleteTask(String input) throws BobException {
+        String[] command = input.split(" ");
+        Integer index = Integer.valueOf(command[1]);
+
+        if (index < 1 || index > taskList.size()) {
+            throw new BobException("Index given is out of range");
+        }
+        formattedPrint("Sure, removing this task!\n" +
+                       getTaskDescription(taskList.get(index - 1)));
+        taskList.remove(index - 1);
+    }
+
+    private void readTaskList() {
+        String flag = "bye";
```

Comment:
can just write: while !input.equals("bye"), dont need a flag variable here

Reply:
I agree. String does not really represent a flag too. So it is better to rename it to something else or use  while !input.equals("bye") 😄